src/include/str_list.h

```diff
@@ -71,29 +71,29 @@ extern void get_str_vec(const std::string& str,
 std::vector<std::string> get_str_vec(const std::string& str,
                                      const char *delims = ";,= \t");
 /**
- * Split **str** into a set of strings, using the ";,= \t" delimiters and output the result in **str_list**.
+ * Split **str** into a set of strings, using the ";,= \t" delimiters and output the result in **str_set**.
  * 
  * @param [in] str String to split and save as Set
- * @param [out] str_list Set modified containing str after it has been split
+ * @param [out] str_set Set modified containing str after it has been split
 **/
 extern void get_str_set(const std::string& str,
-			std::set<std::string>& str_list);
+			std::set<std::string>& str_set);
```

Comment:
i think the right direction is to drop this function and use `for_each_substr()` instead.

Reply:
@tchaikov 
This API has been used by Ceph e.g. https://github.com/ceph/ceph/blob/master/src/rgw/rgw_cors_swift.h#L40

Reply:
yes, that’s why I suggest to use an alternative.

Reply:
@tchaikov 
Do you mean that we need to try to use the optimization in https://github.com/ceph/ceph/pull/18798 and try to remove get_str_set if the code doesn't need get_str_set at all at last?

Reply:
yeah, i don't think we should improve deprecated APIs. and IMHO, the smaller the code base is the less bugs it might carry.

Reply:
I'll try to check the places where it's possible to use ```for_each_substr``` to do the potential optimization since https://github.com/ceph/ceph/pull/18798 says that it could improve 30% performance.
BTW, ```for_each_substr``` is based on ```string_view```. Sometimes, we need to allocate and copy sub-string into another memory space. For example, https://github.com/ceph/ceph/blob/master/src/rgw/rgw_cors_swift.h#L71, we need to store all the sub-string to construct RGWCORSRule. Under these cases, get_str_xxx has good wrapped encapsulation.

Reply:
> Under these cases, get_str_xxx has good wrapped encapsulation.

i disagree,`for_each_substr()` saves 1 extra copy. while in following implementation, we have 2 copies, one in `get_str_set()`, one in following line:

see https://github.com/ceph/ceph/blob/7ffa753be032a3c6da58fc268a56c79c090b4b9b/src/rgw/rgw_cors_swift.h#L44

Reply:
@tchaikov 
Yes. I'll try to remove redundant copy.